code/script.js

```diff
@@ -222,96 +258,139 @@ const generateBoard = () => {
   const { category, value } = currentQuestion
   // Show the correct alert message for different categories
   if (category === 'accessories') {
     if (keep) {
       alert(
         `Yes, the person wears ${value}! Keep all people that wears ${value}`
       )
+      peopleInPlay = peopleInPlay.filter((person) => person[category].includes(value))
     } else {
       alert(
         `No, the person doesn't wear ${value}! Remove all people that wears ${value}`
       )
+      peopleInPlay = peopleInPlay.filter((person) => !person[category].includes(value))
     }
   } else if (category === 'other') {
-    // Similar to the one above
-  } else {
     if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all people with yellow hair"
+      alert(
+        `Yes, the person has ${value}! Keep all people that has ${value}`
+      )
+      peopleInPlay = peopleInPlay.filter((person) => person[category].includes(value))
     } else {
-      // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
+      alert(
+        `No, the person doesn't have ${value}! Remove all people that have ${value}`
+      )
+      peopleInPlay = peopleInPlay.filter((person) => !person[category].includes(value))
     }
-  }
-
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
-
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
-
+  } else if (category === 'hair') {
+    if (keep) {
+      alert(
+        `Yes, the person has ${value}! Keep all people that has ${value}`
+      )
+      peopleInPlay = peopleInPlay.filter((person) => person[category] === value)
+    } else {
+      alert(
+        `No, the person doesn't have ${value}! Remove all people that have ${value}`
+      )
+      peopleInPlay = peopleInPlay.filter((person) => person[category] !== value)
+    }
+  } else if (category === 'eyes') {
+    if (keep) {
+      alert(
+        `Yes, the person has ${value}! Keep all people that has ${value}`
+      )
+      peopleInPlay = peopleInPlay.filter((person) => person[category] === value)
+    } else {
+      alert(
+        `No, the person doesn't have ${value}! Remove all people that have ${value}`
+      )
+      peopleInPlay = peopleInPlay.filter((person) => person[category] !== value)
+    }
+  } 
+  generateBoard()
   // Invoke a function to redraw the board with the remaining people.
 }
 
 // when clicking guess, the player first have to confirm that they want to make a guess.
```

Comment:
I really like (yes i say that alot ;) ) that you save place and have the "peopleInPlay = peopleInPlay.filter((person) => person[category] !== value)" under the alerts, instead of how i did it with a separate if statment. 

Just one thought about the  "readablity" when you have it underneath the alerts, unless you do a comment above this block of code example: -"Here is my if else stament, and after every alert depending on the value the board, it will filter right after with "peopleInPlay = peopleInPlay.filter((person) => person[category] !== value)". :)